enhancements/sig-architecture/15-resourcebasedscheduling/README.md

```diff
@@ -0,0 +1,191 @@
+
+## Summary
+
+The proposed work would enhance the Placement APIs to support scheduling based on the resource usage/capacity of managed clusters.
+
+## Motivation
+
+When scheduling placement, select managed clusters based on cluster resource usage/capacity to prevents placement of workloads on clusters that have insufficient resources to actually run the workloads.
+
+Link to the feature request: https://github.com/open-cluster-management-io/community/issues/52
+
+### User Stories
+
+#### Story 1: User is able to specify resources in placement spec to favor clusters with larger allocatable to capacity ratio of the resources.
+  - User wants to deploy an application on a cluster whose CPU usage is less than others;
+  - User wants to deploy an application on 2 clusters whose allocatable to capacity ratio of both CPU and memory is larger than others;
+
+#### Story 2: User is able to specify resources in placement spec to favor clusters with larger amount of resources allocatable.
+  - User wants to deploy an application on a cluster with largest memory allocatable;
+
+#### Story 3: User is able to create a placement to select clusters based on resource usage, and then keep placement decisions updated according to the changes on cluster resource usage.
+  - A CPU-intensive application would like to be deployed on the cluster with least CPU utilization rate.
+
+#### Story 4: User is able to create a placement to select clusters based on resource usage, and then ignore any resource usage change afterwards to keep the placement decisions pinning.
+  - Some critical service should run on a steady set of clusters, even if the resource usage of those clusters changes;
+
+#### Story 5: Support scheduling based on usage/allocatable of resources other than CPU and Memory. For example,
+  - AI applications are expected to run on clusters with least gpu usage;
+  - An application can be scheduled to cluster with largest available network bandwidth;
+
+### Goals
+
+- Sotry 1 and 2, allow user to select managed clusters based on resource usage/capability with the Placement APIs;
+
+### Non-Goals
+- Balance workload across the fleet based on cluster resource usage;
+- Story 3 and 4 are related to churning policy of placement, and will be covered in a separated enhancement proposal;
+- Story 5 will not be supported until there is a clear requirement;
+
+## Proposal
+
+### 1. Changes on Placement APIs
+Add a field `ClusterResourcePreference` into the Placement spec to allow users to set the cluster resource preference.
+```go
+type PlacementSpec struct {
+	// ... ...
+
+	// ClusterResourcePreference represents the cluster resource preference configuration for
+	// the placement.
+	// This field is ignored when NumberOfClusters in spec is not specified.
+	// +optional
+	ClusterResourcePreference *ClusterResourcePreference `json:"clusterResourcePreference,omitempty"`
```

Comment:
What is the difference between  nil and empty?

Reply:
Here is the difference:
- If nil, it means the placement has no `ClusterResourcePreference` at all;
- if empty, it means the placement has an empty `ClusterResourcePreference`, whose type is `MostAllocatableToCapacityRatio`. Consider `ClusterResourcePreference.ClusterResource` must have at lease one item, an empty `ClusterResourcePreference` is invalid.

```go
 	// +kubebuilder:validation:MinItems:=1
        // +required
 	ClusterResources []ClusterResource `json:"clusterResources"`
```

Reply:
Given steady and balance weighting, it seems like we may want a more generic prioritizing configuration.

Reply:
A generic prioritizing configuration is provided instead of `ClusterResourcePreference`